### PR TITLE
[TypeDeclaration] Register FlipTypeControlToUseExclusiveTypeRector to type-declaration set

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

File: rules/naming/src/VariableRenamer.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 namespace Rector\Naming;
 
+use Closure;
```
Comment (samsonasik) [inline rules/naming/src/VariableRenamer.php:7]: this is false positive as there is already `use PhpParser\Node\Expr\Closure;` means `\Closure` is invalid.

File: rules/nette-code-quality/src/NodeAdding/FunctionLikeFirstLevelStatementResolver.php [hunk 2]
```diff
@@ -57,9 +57,8 @@ public function resolveFirstLevelStatement(Node $node): Node
      */
     private function matchMultiplierClosure(Node $node): ?Closure
     {
-        /** @var Closure|null $closure */
         $closure = $node->getAttribute(AttributeKey::CLOSURE_NODE);
-        if ($closure === null) {
+        if (! $closure instanceof \Closure) {
```
Comment (samsonasik) [inline rules/nette-code-quality/src/NodeAdding/FunctionLikeFirstLevelStatementResolver.php:61]: this is false positive as there is already use PhpParser\Node\Expr\Closure; means \Closure is invalid.

File: rules/naming/src/VariableRenamer.php (hunk 1 above)
Comment (samsonasik) [reply rules/naming/src/VariableRenamer.php]: fixed.

File: rules/nette-code-quality/src/NodeAdding/FunctionLikeFirstLevelStatementResolver.php (hunk 2 above)
Comment (samsonasik) [reply rules/nette-code-quality/src/NodeAdding/FunctionLikeFirstLevelStatementResolver.php]: fixed.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php [hunk 3]
```diff
@@ -125,7 +125,7 @@ private function processConvertToExclusiveType(array $types, Expr $expr, PhpDocI
         /** @var VarTagValueNode $tagValueNode */
         $tagValueNode = $phpDocInfo->getVarTagValueNode();
         $phpDocInfo->removeTagValueNodeFromNode($tagValueNode);
-        return new BooleanNot(new Instanceof_($expr, new FullyQualified($type)));
+        return new BooleanNot(new Instanceof_($expr, new Name($type)));
```
Comment (TomasVotruba) [inline rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php:128]: This should work with FQN, as we use it everywhere in Rector by convention.

The name resolution of `Closure` must be fixed above, probably in `$type`.

Comment (TomasVotruba) [inline rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php:128]: This should work with FQN, as we use it everywhere in Rector by convention.

The name resolution of `Closure` must be fixed above, probably in `$type`.

Comment (TomasVotruba) [inline rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php:128]: This should work with FQN, as we use it everywhere in Rector by convention.

The name resolution of `Closure` must be fixed above, probably in `$type`.

Comment (samsonasik) [reply rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php:128]: The type name returns actual value like in the doc, eg:

```php
@var stdClass|null
``` 

The types[0] will be 'stdClass'

and for :


```php
@var \stdClass|null
``` 

The types[0] will be '\stdClass'

any specific type converter for it?

Comment (TomasVotruba) [reply rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php:128]: I think `StaticTypeMapper` would do here.


Proably `mapPHPStanPhpDocTypeToPHPStanType()` method

Comment (samsonasik) [reply rules/type-declaration/src/Rector/Identical/FlipTypeControlToUseExclusiveTypeRector.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you, this is a huge improvement in our coding :+1:

**Review by TomasVotruba — APPROVED** [review]

**Pull request merged**